AmericanMedicalAssociation/ama-style-guide-2
# [EWL-4514] SG2 | Fix Visual Regressions in Promo pattern variants and add Pattern Info

## Ticket(s)

**Jira Ticket**
- [EWL-4514: SG2 | Fix Visual Regressions in Promo pattern variants and add Pattern Info](https://issues.ama-assn.org/browse/EWL-4514)

## Description
**This PR is the SG2 counterpart to https://github.com/AmericanMedicalAssociation/ama-d8/pull/274. It needs to be merged before that PR is merged. These PRs should be tested and merged alongside one another (as in, one should not be merged without the other).**

- Fixes the Promo pattern and its variants so that they align with the project requirements. 
- Adds/corrects the Pattern Info for these patterns.

## To Test
`gulp serve`

- [ ] When I click "molecules" in the SG2 menu
--- Then I should see "Promo"
--- And when I click "Promo" 
--- Then I should see that the styling looks correct according to the attached screenshot
--- And I should see that the pattern title is "Promo" and NOT "Promotion"
--- And when I click "Pattern Info" in the Gear Icon menu
--- Then I should see a list of the pattern's variants
--- And I should see that the info in the `.md` file is correct (.json data structure, etc.)

- [ ] And when I click  "molecules" in the SG2 menu again
--- Then I should see "Promo with Border"
--- And when I click "Promo with Border"
--- Then I should see that the styling looks correct according to the attached screenshot
--- And when I click "Pattern Info" in the Gear Icon menu
--- Then I should see information about the pattern and a link back to the "Promo" base pattern

- [ ] And when I click "molecules" in the SG2 menu again
--- Then I should see "Promo with CTA"
--- And when I click "Promo with CTA"
--- Then I should see that the styling looks correct according to the attached screenshot
--- And I should see a full width button is being displayed with style *secondary* (transparent button with border)
--- And when I click "Pattern Info" in the Gear Icon menu
--- Then I should see information about the pattern and a link back to the "Promo" base pattern

- [ ] And when I click "molecules" in the SG2 menu again
--- Then I should see "Promo with Image"
--- And when I click "Promo with Image"
--- Then I should see that the styling looks correct according to the attached screenshot
--- And when I click "Pattern Info" in the Gear Icon menu
--- Then I should see information about the pattern and a link back to the "Promo" base pattern

- [ ] And when I click "pages" in the SG2 menu
--- And when I click "topic"
--- Then I should see a "Promo with CTA" in the left sidebar under Tools

## Visual Regressions
I added scenarios to `backstop.json` for the Promo, Promo with Border, and Promo with CTA variants. Some of the references images did not generate, so I may need someone to run `backstop test` when pulling down the branch and then use `backstop approve` to commit the missing reference images.

Backstop test run on this branch did not show any unexpected regressions in existing tests for patterns.

## Relevant Screenshots/GIFs
Promo 
![screen shot 2018-02-22 at 3 28 11 pm](https://user-images.githubusercontent.com/12160398/36565083-073ae254-17e5-11e8-95e5-fa0cb064d0ce.png)

Promo with style Border
![promoborder](https://user-images.githubusercontent.com/12160398/36448037-e024e7cc-164b-11e8-81b8-75b2448c2bfd.png)

Promo with style Border and visible CTA
![promobordercta](https://user-images.githubusercontent.com/12160398/36448038-e032310c-164b-11e8-9dbd-0aeba60b36af.png)

Promo with style Background
![promobackground](https://user-images.githubusercontent.com/12160398/36448041-e057433e-164b-11e8-96bc-d24a00db61a7.png)

Promo with style Background and visible CTA
![promobackgroundcta](https://user-images.githubusercontent.com/12160398/36448039-e03f1890-164b-11e8-88ce-6244fb5b8c0f.png)

Promo with Image
![screen shot 2018-02-22 at 3 29 44 pm](https://user-images.githubusercontent.com/12160398/36565160-409a912a-17e5-11e8-8f68-8bac2f4bd752.png)



## Remaining Tasks
- [x] Make sure this isn't broken in d8 and all the variants still work - see https://github.com/AmericanMedicalAssociation/ama-d8/pull/274
- [x] Add "Promo" to [The Glossary](https://issues.ama-assn.org:8446/confluence/display/DTD/Glossary?src=contextnavpagetreemode) 

## Additional Notes
This PR is the SG2 counterpart to https://github.com/AmericanMedicalAssociation/ama-d8/pull/274 - that PR should be merged alongside this PR. 

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)
